sdk/attestation/azure-security-attestation/inc/azure/attestation/attestation_administration_client.hpp

```diff
@@ -44,6 +44,8 @@ namespace Azure { namespace Security { namespace Attestation {
    *
    */
   class AttestationAdministrationClient final {
+    friend class AttestationAdministrationClientFactory;
```

Comment:
Given this pattern will set a precedence of what factory-based client creation looks like, can you help me understand why the client class needs to have the factory be a friend?

Reply:
We'd want to add guidance in the guidelines on when/why to use this pattern for other SDKs and the client/factory class relationship.

Reply:
In the factory pattern, the client class's constructor is private. As long as the factory is a separate class, the factory must be a friend.

And the user experience for a separate factory class is vastly better than the experience for a factory method.

Reply:
> And the user experience for a separate factory class is vastly better than the experience for a factory method.

I am curious to know why a separate class is a clear winner here, from a usability perspective.